### PR TITLE
Add version information for arrai binary

### LINKED ISSUE #246: Version flag for binary
Please **do not post any internal, closed source snippets** on this public issue tracker!

### Current arrai Version
It would be nice to be able to check for the current build version of arrai. Either by date or commit hash.

### Use-cases
It would be great for debugging as well as for issues. E.g. Provide arrai build version for bug reproducability.

### Attempted Solutions
```
$ arrai v
build version: 834131ab82c7878c8e8
etc
```

Needs to work in CI and the Makefile. The Makefile option is important because the main [README](https://github.com/arr-ai/arrai#install) recommends installing from source on Unix-like platforms.

For Makefile versioning, we also need to mark builds as dirty if the git repo isn't clean. There are three scenarios to consider:

1. Clean and version-tagged
    ```bash
    $ arrai v
    arrai v0.123.0 darwin/amd64
    ```
1. Clean but no version tag:
    ```bash
    $ arrai v
    arrai 751a975944edadf56a2558a8a21eebcfd47fc52e~3 = v0.123.0 darwin/amd64
    ```
1. Dirty (local working tree changes, including untracked files) and version-tagged:
    ```bash
    $ arrai v
    arrai DIRTY-v0.123.0 darwin/amd64
    ```
1. Dirty (local working tree changes, including untracked files) but no version tag:
    ```bash
    $ arrai v
    arrai DIRTY-751a975944edadf56a2558a8a21eebcfd47fc52e~3 = v0.123.0 darwin/amd64
    ```

Do likewise for `arrai v -v`.

`~3 = v0.123.0` indicates that v0.123.0 is three commits behind HEAD.

### PR DESCRIPTION
**Pull request by ericzhang6222**
Changes proposed in this pull request:
- Add version information for arrai binary

Test link https://github.com/ericzhang6222/arrai/releases

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

**Comment by ericzhang6222** [comment]
Output sample:
Build:
  Version      : 0.28.0
  Git Commit   : e8a7677
  Date         : 2020-06-05T02:01:05Z
  Go Version   : go1.14.4 linux/amd64
  OS           : darwin/amd64
Runtime:
  GOOS/GOARCH  : darwin/amd64
  Go Version   : go1.14.4

Test link https://github.com/ericzhang6222/arrai/runs/740765620

**Comment by ericzhang6222** [comment]
> Too much information. Add a `--verbose|-v` flag to report roughly as it is now (see below example), but when not specified, just report version and platform:
> 
> ```shell
> $ arrai version
> arrai 0.123.0 linux/amd64
> $ arrai version -v
> Version    : 0.28.0
> Git commit : 5cebabb229a6ff9cb127f7ecd4ae32f85b7b019d
> Date       : 2020-06-05T02:01:05Z
> OS/arch    : darwin/amd64
> Go version : go1.14.4 linux/amd64
> ```

Fixed, current output:
l68f728fb99c8:arrai_0.29.0_darwin-amd64 ezhang$ ./arrai v
```
Version    : 0.29.0
OS/arch    : darwin/amd64
```

l68f728fb99c8:arrai_0.29.0_darwin-amd64 ezhang$ ./arrai v -v
```
Version    : 0.29.0
Git commit : 49c19b6a682b4ebe6c994ab3c54b60e2b8b2e453
Date       : 2020-06-05T09:42:22Z
OS/arch    : darwin/amd64
Go version : go1.14.4 linux/amd64
```

**Comment by marcelocantos** [comment]
I prefer this single-line format for the simple version:
```shell
arrai 0.123.0 linux/amd64
```

There's plenty of precedent for this:
```
$ go version
go version go1.14.3 linux/amd64
$ git version
git version 2.24.1
$ node --version
v10.15.2
$ python --version
Python 2.7.17
$ ruby --version
ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-linux-gnu]
$ erl -version
Erlang (SMP,ASYNC_THREADS) (BEAM) emulator version 10.4.4
$ mono-csc --version
Mono C# compiler version 5.18.0.240
$ sqlite3 --version
3.29.0 2019-07-10 17:32:03 fc82b73eaac8b36950e527f12c4b5dc1e147e6f4ad2217ae43ad82882a88alt1
$ rstudio --version
1.3.959
```

There are numerous counterexamples, but they seem to fall mostly into the category of copyright notices on GNU FSF applications, e.g.:

```
$ c++ --version
c++ (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

...a precedent I do _not_ want to follow.

File: cmd/arrai/version.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func version(c *cli.Context) error {
+	if c.Bool("verbose") {
+		fmt.Printf("Version    : %s\n", Version)
+		fmt.Printf("Git commit : %s\n", GitFullCommit)
+		fmt.Printf("Date       : %s\n", BuildDate)
+		fmt.Printf("OS/arch    : %s\n", BuildOS)
+		fmt.Printf("Go version : %s\n", GoVersion)
+	} else {
+		fmt.Printf("Version    : %s\n", Version)
+		fmt.Printf("OS/arch    : %s\n", BuildOS)
```
Comment (marcelocantos) [inline cmd/arrai/version.go:31]: I prefer this single-line format for the simple version:
```shell
arrai 0.123.0 linux/amd64
```

There's plenty of precedent for this:
```
$ go version
go version go1.14.3 linux/amd64
$ git version
git version 2.24.1
$ node --version
v10.15.2
$ python --version
Python 2.7.17
$ ruby --version
ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-linux-gnu]
$ erl -version
Erlang (SMP,ASYNC_THREADS) (BEAM) emulator version 10.4.4
$ mono-csc --version
Mono C# compiler version 5.18.0.240
$ sqlite3 --version
3.29.0 2019-07-10 17:32:03 fc82b73eaac8b36950e527f12c4b5dc1e147e6f4ad2217ae43ad82882a88alt1
$ rstudio --version
1.3.959
```

There are numerous counterexamples, but they seem to fall mostly into the category of copyright notices on GNU FSF applications, e.g.:

```
$ c++ --version
c++ (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

...a precedent I do _not_ want to follow.

Comment (ericzhang6222) [reply cmd/arrai/version.go]: Current output for simple version:
```
arrai version 0.30.0 darwin/amd64
```

**Pull request merged**